### PR TITLE
refactor: use default sql builders to parse UPDATE and DELETE

### PR DESCRIPTION
**Pull request by mikezaschka**

### CODE REVIEW

File: __tests__/lib/pg/service.test.js [hunk 1]
```diff
@@ -138,10 +138,36 @@ describe('OData to Postgres dialect', () => {
...
+      expect(getResponse.body).toEqual(expect.objectContaining({ name: 'Changed name', ibu: 10 }))
+    })
+  })
+
   describe('odata: DELETE -> sql: DELETE', () => {
     test('odata delete ', async () => {
       const response = await request.delete(`/beershop/Beers/9e1704e3-6fd0-4a5d-bfb1-13ac47f7976b`).send()
       expect(response.status).toStrictEqual(204)
+
+      // REVISIT: Works and test does not fail but console contains express error (404). How to solve this?
```
Comment (vobu) [inline __tests__/lib/pg/service.test.js:166]: don't really understand the REST-type query here...probably left-over to be deleted? ;)

File: lib/pg/query/delete.js [hunk 2]
```diff
@@ -1,28 +1,12 @@
-/**
- * Handle delete calls to postgres
+const { processCQN } = require('../execute')
+
+/*/**
```
Comment (vobu) [inline lib/pg/query/delete.js:3]: would you mind fixing the JSDoc here
/** -> start
dbc is of type PoolClient
cqn probably String|Object?

File: lib/pg/query/update.js [hunk 3]
```diff
@@ -1,37 +1,12 @@
-const format = require('pg-format')
+const { processCQN } = require('../execute')
 
 /**
- * Handle update calls to postgres
  *
  * @param {*} dbc
```
Comment (vobu) [inline lib/pg/query/update.js:5]: dbc is of type PoolClient
cqn probably String|Object?

**Review by vobu — CHANGES_REQUESTED** [review]
only form thingies, very cool that the mapping to the cds core cqn handler works out of the box!

**Comment by vobu** [comment]
oh, and check-off that todo on `README.md` :)

File: __tests__/lib/pg/service.test.js (hunk 1 above)
Comment (mikezaschka) [reply __tests__/lib/pg/service.test.js:166]: And I don't understand your comment. ;-)
What do you mean by REST-type query?

Comment (vobu) [reply __tests__/lib/pg/service.test.js:166]: the commented out part with "REVISIT"

Comment (mikezaschka) [reply __tests__/lib/pg/service.test.js:166]: Still confused. :-)
I wanted to check if the record is really gone in the db. But when you fire the next GET request, the odata server responds with a 404 and throws an exception that is visible in the terminal output, but does not affect the test results.

Comment (vobu) [reply __tests__/lib/pg/service.test.js:196]: ah, ok, my bad. now I get it (coffee helped!). let me try out a thingie or two and get back to you on why the subsequent GET doesn't resolve to the desired 404.

Comment (vobu) [reply __tests__/lib/pg/service.test.js:196]: this seems to work fine for me:
```js
  test('odata: delete single beer -> sql: delete record', async () => {
      const guid = '9e1704e3-6fd0-4a5d-bfb1-13ac47f7976b'
      const response = await request.delete(`/beershop/Beers(${guid})`)
      expect(response.status).toStrictEqual(204)

      // make sure the deleted beer doesn't exist anymore
      const subsequentResponse = await request.get(`/beershop/Beers(${guid})`)
      expect(subsequentResponse.status).toStrictEqual(404)
    })
```
can you double-check please?

Comment (mikezaschka) [reply __tests__/lib/pg/service.test.js:196]: This pops up on my machine. The test itself does not break, but I get an ugly terminal output.
If that's not happening on your machine or if it is fine, that let's merge this.

image

Comment (vobu) [reply __tests__/lib/pg/service.test.js]: jep, same here - and that's fine for me given that it's the @sap/cds odata server logging the 404

**Review by vobu — APPROVED** [review]

**Pull request merged**